### PR TITLE
docs: improve the description of pattern and allowedCharPattern

### PR DESCRIPTION
**Pull request by vursen**
The PR aims to improve the description of the pattern and allowedCharPattern Text Field features to better reflect the difference between them.

Fixes https://github.com/vaadin/docs/issues/1869

### CODE REVIEW

File: articles/components/text-field/index.asciidoc [hunk 1]
```diff
@@ -161,12 +163,13 @@ include::{root}/src/main/java/com/vaadin/demo/component/textfield/TextFieldPatte
...
 --
 
-This example invalidates any value that doesn't consist exclusively of numbers, parenthesis, dashes, and the plus sign. In cases where the format requirements may not be clear to the user, it's recommended to provide this information, for example by using a <<../input-fields#helper,helper>>.
+In cases where the format requirements may not be clear to the user, it's recommended to provide this information, for example by using a <<../input-fields#helper,helper>>.
+
+== Preventing Entry of Invalid Characters
 
-[role="since:com.vaadin:vaadin@V23.2"]
-=== Preventing Invalid Input
+Characters that you enter into Text Field can be restricted using a regular expression. When enabled, the user can't enter characters that match the regular expression. If they try to enter such characters, the component gives visual feedback that the input was prevented.
```
Comment (vursen) [inline articles/components/text-field/index.asciidoc:170]:
```suggestion
Characters that you enter into Text Field can be restricted using a single-character regular expression. When enabled, the user can't enter characters that match the regular expression. If they try to enter such characters, the component gives visual feedback that the input was prevented.
```

File: articles/components/text-field/index.asciidoc [hunk 2]
```diff
@@ -161,12 +163,13 @@ include::{root}/src/main/java/com/vaadin/demo/component/textfield/TextFieldPatte
...
 --
 
-This example invalidates any value that doesn't consist exclusively of numbers, parenthesis, dashes, and the plus sign. In cases where the format requirements may not be clear to the user, it's recommended to provide this information, for example by using a <<../input-fields#helper,helper>>.
+In cases where the format requirements may not be clear to the user, it's recommended to provide this information, for example by using a <<../input-fields#helper,helper>>.
+
+== Preventing Entry of Invalid Characters
 
-[role="since:com.vaadin:vaadin@V23.2"]
-=== Preventing Invalid Input
+Characters that you enter into Text Field can be restricted using a regular expression. When enabled, the entry of characters not matching the regular expression is prevented and gives the field a short shake.
```
[the remark this answers is not in the document]
Comment (vursen) [reply articles/components/text-field/index.asciidoc]: Addressed in the recent version. Let me know if you think we should specifically highlight that full regular expressions won't work.

File: articles/components/text-field/index.asciidoc [hunk 3]
```diff
@@ -146,7 +146,9 @@ In cases where the length requirements may not be clear to the user, it's recomm
 
 == Pattern
 
-A custom validation pattern can be defined (using regular expressions) to validate input.
+Pattern is a regular expression that Text Field's value is matched against during validation. Any value that doesn't match the pattern invalidates the field.
+
+The following examples invalidates any value that doesn't consist exclusively of numbers, parentheses, dashes, and the plus sign:
```
Comment (rolfsmeds) [inline articles/components/text-field/index.asciidoc:151]:
```suggestion
The following example invalidates any value that doesn't consist exclusively of numbers, parentheses, dashes, and the plus sign:
```

File: articles/components/text-field/index.asciidoc [hunk 4]
```diff
@@ -146,7 +146,9 @@ In cases where the length requirements may not be clear to the user, it's recomm
 
 == Pattern
 
-A custom validation pattern can be defined (using regular expressions) to validate input.
+Pattern is a regular expression that Text Field's value is matched against during validation. Any value that doesn't match the pattern invalidates the field.
```
Comment (rolfsmeds) [inline articles/components/text-field/index.asciidoc:149]:
```suggestion
The pattern is a regular expression that Text Field's complete value is matched against during validation. Any value that doesn't match the pattern invalidates the field.
```

File: articles/components/text-field/index.asciidoc [hunk 5]
```diff
@@ -161,12 +163,13 @@ include::{root}/src/main/java/com/vaadin/demo/component/textfield/TextFieldPatte
...
 --
 
-This example invalidates any value that doesn't consist exclusively of numbers, parenthesis, dashes, and the plus sign. In cases where the format requirements may not be clear to the user, it's recommended to provide this information, for example by using a <<../input-fields#helper,helper>>.
+In cases where the format requirements may not be clear to the user, it's recommended to provide this information, for example by using a <<../input-fields#helper,helper>>.
+
+== Preventing Entry of Invalid Characters
 
-[role="since:com.vaadin:vaadin@V23.2"]
-=== Preventing Invalid Input
+Characters that the user enters into Text Field can be restricted using a single-character regular expression. When enabled, the user can only enter characters that match the regular expression. If they try to enter any other characters, the component gives visual feedback that the input was prevented.
```
Comment (rolfsmeds) [inline articles/components/text-field/index.asciidoc:170]:
```suggestion
Individual characters that the user enters into Text Field can be restricted using a _single-character_ regular expression. When enabled, the user can only enter characters that match the regular expression. If they try to enter any other characters, the component gives visual feedback that the input was prevented.
```

**Review by rolfsmeds — CHANGES_REQUESTED** [review]
Made some minor adjustments: I feel this is good enough for now – we can make it even more explicit if misunderstandings continue :)

**Comment by jouni** [comment]
Merged based on Rolf’s comment, even though his review requested changes.

**Pull request merged**